lib/web/databases.go

```diff
@@ -0,0 +1,148 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/tlsutils"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/web/ui"
+)
+
+// createDatabaseRequest contains the necessary basic information to create a database.
+// Database here is the database resource, containing information to a real database (protocol, uri)
+type createDatabaseRequest struct {
+	Name     string     `json:"name,omitempty"`
+	Labels   []ui.Label `json:"labels,omitempty"`
+	Protocol string     `json:"protocol,omitempty"`
+	URI      string     `json:"uri,omitempty"`
+}
+
+func (r *createDatabaseRequest) checkAndSetDefaults() error {
+	if r.Name == "" {
+		return trace.BadParameter("missing database name")
+	}
+
+	if r.Protocol == "" {
+		return trace.BadParameter("missing protocol")
+	}
+
+	if r.URI == "" {
+		return trace.BadParameter("missing uri")
+	}
+
+	return nil
+}
+
+// handleDatabaseCreate creates a database's metadata.
+func (h *Handler) handleDatabaseCreate(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	var req *createDatabaseRequest
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := req.checkAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	labels := make(map[string]string)
+	for _, label := range req.Labels {
+		labels[label.Name] = label.Value
+	}
+
+	database, err := types.NewDatabaseV3(
+		types.Metadata{
+			Name:   req.Name,
+			Labels: labels,
+		},
+		types.DatabaseSpecV3{
+			Protocol: req.Protocol,
+			URI:      req.URI,
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := clt.CreateDatabase(r.Context(), database); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return ui.MakeDatabase(database), nil
+}
+
+// updateDatabaseRequest contains some updatable fields of a database resource.
+type updateDatabaseRequest struct {
+	CACert string `json:"ca_cert,omitempty"`
+}
+
+func (r *updateDatabaseRequest) checkAndSetDefaults() error {
+	if r.CACert == "" {
+		return trace.BadParameter("missing ca_cert field")
+	}
+
+	if _, err := tlsutils.ParseCertificatePEM([]byte(r.CACert)); err != nil {
+		return trace.BadParameter("invalid x509 PEM certificate in ca_cert")
```

Comment:
Nit: I'd tweak the error message as the user wouldn't really know what `ca_cert` is.
```suggestion
		return trace.BadParameter("could not parse provided CA as X.509 PEM certificate")
```